hudi-common/src/main/java/org/apache/hudi/common/model/HoodieAvroIndexedRecord.java

```diff
@@ -243,7 +292,7 @@ public ByteArrayOutputStream getAvroBytes(Schema recordSchema, Properties props)
   @Override
   protected final void writeRecordPayload(IndexedRecord payload, Kryo kryo, Output output) {
     // NOTE: We're leveraging Spark's default [[GenericAvroSerializer]] to serialize Avro
-    Serializer<GenericRecord> avroSerializer = kryo.getSerializer(GenericRecord.class);
+    Serializer<SerializableIndexedRecord> avroSerializer = kryo.getSerializer(SerializableIndexedRecord.class);
```

Comment:
note to register `SerializableIndexedRecord` with Kryo. in the right places. Please resolve if its taken care of

Reply:
Ack, updated